openverse_catalog/dags/common/dag_factory.py

```diff
@@ -22,19 +32,64 @@
     "on_failure_callback": slack.on_failure_callback,
 }
 DATE_RANGE_ARG_TEMPLATE = "{{ macros.ds_add(ds, -%s) }}"
+XCOM_PULL_TEMPLATE = "{{ ti.xcom_pull(task_ids='%s', key='%s') }}"
+
+
+def _push_output_paths_wrapper(
+    func: Callable,
+    media_types: List[str],
+    ti: TaskInstance,
+    args: Sequence = None,
+):
+    """
+    Run the provided callable after pushing the calculated output directories
+    for each media store to XComs. Output locations are pushed under keys with
+    the format `<media-type>_tsv`. This is a temporary workaround due to the nature
+    of the current provider scripts. Once
+    https://github.com/WordPress/openverse-catalog/issues/229 is addressed and the
+    provider scripts are refactored into classes, this wrapper can either be updated
+    or the XCom pushing can be moved into the provider initialization.
+    """
+    args = args or []
+    logger.info("Pushing available store paths to XComs")
+    module = inspect.getmodule(func)
+    stores = {}
+
+    # Stores exist at the module level, so in order to retrieve the output values we
+    # must first pull the stores from the module.
+    for media_type in media_types:
+        if not (store := getattr(module, f"{media_type}_store", None)):
+            continue
+        stores[media_type] = store
+
+    if len(stores) != len(media_types):
+        raise ValueError(
+            f"Expected a store of the following types in {module.__name__} "
+            f"but none were found: {media_types}"
```

Comment:
It might be useful to log which stores were missing, or else change the wording slightly for clarity (`none were found` -> `not all were found`?).

Reply:
How is this?
![image](https://user-images.githubusercontent.com/10214785/154574265-0db18668-c2b4-46ec-bd4c-6dd684be0768.png)